src/Persistence/MassTransit.EntityFrameworkCoreIntegration/MassTransit.EntityFrameworkCoreIntegration.csproj

```diff
@@ -14,7 +14,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-      <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="2.2.6" />
+      <PackageReference Include="Microsoft.Data.SqlClient" Version="1.1.0" />
+      <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.1.0" />
```

Comment:
Why we need this reference?

Reply:
There's SQL Server specific code in here. It would be a lot more work to make this database-agnostic (and relying on `Microsoft.EntityFrameworkCore.Relational`). If that's what you mean?

Reply:
right now EF Core lib is related to SQLServer but if I want PostgreSQL?
I know that was before.. But wondering if we could do it better
For eg, we could have table name inside `SagaClassMap` but if we have custom `DbContext` it won't work

Reply:
Yeah, try bumping it to the Relational package, then it would be db-agnostic. The only thing you will need to do is update [this line](https://github.com/MassTransit/MassTransit/blob/b49e9ddc60c73cdf894cfc79a6c7f22a90da6b6b/src/Persistence/MassTransit.EntityFrameworkCoreIntegration/Saga/EntityFrameworkSagaRepository.cs#L104). There is probably a base type, you could use that instead. There's a unit test for that, so if it still passes, then that should be acceptable.

Reply:
I had a look at the EF core, the default implementation returns a `NoopExecutionStrategy`, it appears we can just use that regardless and get rid of the condition.

Reply:
There's another thing we'd need to fix to make it work with multiple backends, it seems there's code to [detect deadlocks](https://github.com/MassTransit/MassTransit/blob/b49e9ddc60c73cdf894cfc79a6c7f22a90da6b6b/src/Persistence/MassTransit.EntityFrameworkCoreIntegration/Saga/EntityFrameworkSagaRepository.cs#L407) which obviously uses `Microsoft.Data.SqlClient` directly. It seems to only be used to [avoid a transaction rollback](https://github.com/MassTransit/MassTransit/blob/b49e9ddc60c73cdf894cfc79a6c7f22a90da6b6b/src/Persistence/MassTransit.EntityFrameworkCoreIntegration/Saga/EntityFrameworkSagaRepository.cs#L357) in the case of a deadlock. I suppose this is for performance reasons. We have a few options:

1) Get rid of that code and rollback anyway.
2) Always return false if we don't have a `SqlException`, so the detection will only work on SQL Server.
3) Leave the code SQL Server specific.

Ideas?

Reply:
I think if we look for the type `ExecutionStrategy` instead, then that [should work](https://github.com/aspnet/EntityFrameworkCore/blob/master/src/EFCore.SqlServer/SqlServerRetryingExecutionStrategy.cs).

I see the factory you are looking at here with [Noop](https://github.com/aspnet/EntityFrameworkCore/blob/master/src/EFCore.Relational/Storage/RelationalExecutionStrategyFactory.cs). So if NoopExecutionStrategy is returned, `if (execStrategy isExecutionStrategy)` should skip and process normally.

Reply:
Pushed database agnostic implementation. Removed checks for deadlocks that avoided rollbacks, if anyone has a good reason for them to remain there then I think we'll have to stick to MSSQL-only and revert this commit. Cleaned up some exception handling in the process.

Tests green.

Reply:
Regarding the Deadlock detection. I'm not sure the correct answer, but my guess would be that the majority of people using EfCore are MsSql. So leaving the deadlock detection (and dependency on the `Microsoft.Data.SqlClient` package), is probably the best for the users.

We can always re-visit if we get complaints from people about this package being used unnecessarily.

Reply:
I jumped the gun there!